client/src/sagas/apartments.js

```diff
@@ -0,0 +1,47 @@
+import { takeLatest, call, put} from 'redux-saga/effects';
+import NodeGeocoder from 'geocoder';
+
+import { APARTMENTS_REQUEST, APARTMENTS_SUCCESS, APARTMENTS_FAILURE } from '../constants/apartments';
+
+import fetchApartmentsApi from '../api/apartments';
+
+// Used to map the response of the API request for the apartments
+const mapApartmentsResponse = (apartmentRecords) => {
+  const markers = [];
+  return apartmentRecords.map(record => {
+    return NodeGeocoder.geocode(record.street_address, (err, geoCodedAddress) => {
+      const location = geoCodedAddress.results[0].geometry.location
+      return markers.push({
+        key: record.id,
+        position: {
+          lat: location.lat,
+          lng: location.lng
+        },
+        defaultAnimation: 2
+      });
+    });
+  });
+};
+
+function* apartmentsFlow(action) {
```

Comment:
The generator stuff is super cool, but it means you can't stop showing up until someone else learns it! You've made yourself essential. ;)